pkg/image/resolve.go

```diff
@@ -145,13 +145,27 @@ func addSourceToImage(imagesSet map[string]map[string]bool, image string, source
 	}
 }
 
-func walkthroughMap(inputMap map[interface{}]interface{}, walkFunc func(map[interface{}]interface{})) {
-	walkFunc(inputMap)
-	for _, value := range inputMap {
-		if v, ok := value.(map[interface{}]interface{}); ok {
-			walkthroughMap(v, walkFunc)
+func walkthroughMap(data interface{}, walkFunc func(map[interface{}]interface{})) {
+	inputMap, isMap := data.(map[interface{}]interface{})
+	if isMap {
+		// Run the walkFunc on the root node and each child node
+		walkFunc(inputMap)
+		for _, value := range inputMap {
+			walkthroughMap(value, walkFunc)
 		}
+		return
+	}
+	// Check if data is a list
+	inputList, isList := data.([]interface{})
+	if isList {
+		// Run the walkFunc on each element in the root node, ignoring the root itself
+		for _, elem := range inputList {
+			walkthroughMap(elem, walkFunc)
+		}
+		return
 	}
+	// Element is neither a map nor a list
+	return
```

Comment:
nit for fewer `return`s:
```suggestion
	if inputMap, isMap := data.(map[interface{}]interface{}); isMap {
		// Run the walkFunc on the root node and each child node
		walkFunc(inputMap)
		for _, value := range inputMap {
			walkthroughMap(value, walkFunc)
		}
	} else if inputList, isList := data.([]interface{}); isList {
		// Run the walkFunc on each element in the root node, ignoring the root itself
		for _, elem := range inputList {
			walkthroughMap(elem, walkFunc)
		}
	}
```